### ISSUE TITLE
PHP hangs when parent and child inheritance is not setted properly.

### ISSUE THREAD

**Issue by darron1217** [jonspalmer/single-table-inheritance#66]
PHP hangs when parent and child inheritance is not setted properly.

For example, the class `MiniCar` is not proper inheritance. But it never emits error, just hang PHP which makes hard to debug.
```
use Nanigans\SingleTableInheritance\SingleTableInheritanceTrait;

class Vehicle extends Model
{
  use SingleTableInheritanceTrait;

  protected $table = "vehicles";

  protected static $singleTableTypeField = 'type';

  protected static $singleTableSubclasses = [Car::class];
}

class Car extends Vehicle
{
  protected static $singleTableType = 'car';

  protected static $singleTableSubclasses = [MiniCar::class];
}

// This is wrong, but it never emits error.
class MiniCar extends Vehicle
{
  protected static $singleTableType = 'minicar';
}
```

```
// This will lead you to infinite execution of PHP script.
Vehicle::getSingleTableTypeMap();
```

Is there a way to emit errors in this situation?

**Comment by jonspalmer** [comment]
I suspect it's possible. PR would be welcomed if you'd like to take a stab at it.

**Comment by darron1217** [comment]
@jonspalmer 
I created PR :)

**Comment by jonspalmer** [comment]
Fixed by #67
